ios/Runner/AppDelegate.swift

```diff
@@ -37,4 +45,22 @@ let apnTokenKey = "apnToken"
         // Save the token to use in the future
         UserDefaults.standard.set(token, forKey: apnTokenKey)
     }
+    
+    override func userNotificationCenter(_ center: UNUserNotificationCenter,
+                                         willPresent notification: UNNotification,
+                                         withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        let userInfo = notification.request.content.userInfo
+        twakeApnChannel?.invokeMethod("willPresent", arguments: userInfo)
+        let isRemoteNotification = userInfo["aps"] != nil
+        // Hide remote noti when in foreground, decrypted noti will show by dart
+        completionHandler(isRemoteNotification ? [] : [.alert, .badge, .sound])
+    }
+
+    override func userNotificationCenter(_ center: UNUserNotificationCenter,
+                                         didReceive response: UNNotificationResponse,
+                                         withCompletionHandler completionHandler: @escaping () -> Void) {
+        let userInfo = response.notification.request.content.userInfo
+        twakeApnChannel?.invokeMethod("didReceive", arguments: userInfo)
+        completionHandler()
```

Comment:
no need to call `completionHandler` here right ?

Reply:
We need it
**completionHandler**
The block to execute when you have finished processing the user’s response. You must execute this block at some point after processing the user's response to let the system know that you are done. The block has no return value or parameters.
https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate/1649501-usernotificationcenter

Reply:
IMO: You should write ADR for this ticket